### PR TITLE
Port Rust-g DLL and use it for logging

### PR DESCRIPTION
**Pull request by Sindorman**
As requested by @Arrow768 this PR ports rust-g [DLL](https://github.com/tgstation/rust-g) from TG, here is a [PR](https://github.com/tgstation/tgstation/pull/36858/files) that was used as reference. This should reduce CPU usage used by byond file IO.

### CODE REVIEW

File: code/_helpers/logging.dm [hunk 1]
```diff
@@ -1,5 +1,6 @@
 //print an error message to world.log
-
+#define RUST_G "rust_g" // Using Rust g dll to log faster with less CPU usage.
```
Comment (skull132) [inline code/_helpers/logging.dm]: Undef these two at the end of file. They shouldn't be leaking.

**Comment by Sindorman** [comment]
I added changelog just in case. Since changes are sort of visible.

File: TGS3.json [hunk 2]
```diff
@@ -18,5 +18,6 @@
     ],
     "dlls": [
       "libmysql.dll"
```
Comment (Arrow768) [inline TGS3.json]: Invalid json.
Missing the , after "libmysql.dll

Comment (Arrow768) [reply TGS3.json]: Also, space vs tab indentation.
Make sure to use space for the new line you added.

**Comment by Sindorman** [comment]
- Updated - each individual job run compile script

- compiling scrip compiles only if job's cache doesn't have library pre-compiled.

File: scripts/rust_g.sh [hunk 3]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+
+test_rust=$(ls $HOME/rust_g-$RUST_G_VERSION | grep "rust_g")
```
Comment (skull132) [inline scripts/rust_g.sh]: Technically `if [ ! -f $HOME/rust_g-$RUST_G_VERSION/rust_g ]; then` is a more appropriate check. But ye.

**Pull request merged**